packages/eslint-plugin/tests/rules/restrict-plus-operands.test.ts

```diff
@@ -283,5 +285,25 @@ var foo = pair + pair;
         },
       ],
     },
+    {
+      code: `var foo = 1n; foo + 1`,
```

Comment:
This is already an error throw by TypeScript. Why you need a lint rule for it?

https://www.typescriptlang.org/play/index.html#src=var%20foo%20%3D%201n%3B%20foo%20%2B%201

Reply:
@mohsen1 , good point. Maybe it's really an overhead and this rule makes sense only for strings and numbers.

I've implemented this to fix issue #309, but you're right that TypeScript does the same check already.

@novemberborn , @j-f1 , what do you think?

Reply:
Ah yes, it would do that. However the rule then needs to ensure either operand is actually a string or number before complaining (or `any` / `unknown`). The problem in #309 was that it complained about `bigint` even though bigints can be summed.

Reply:
Agree, I'd like to propose the next behaviour for this rule:

`(any or unknown) + (string or number or bigint)` - rule triggers an error, as it does now
`all other cases` - let TS decide what error to show

Reply:
That seems reasonable. You may want to pull that suggestion into its own issue for wider discussion, if you haven't already.

Reply:
I made a table to show the cases this rule should cover in my opinion:

|  +  | number | string | BigInt | any   | unknown |
|---------|--------|--------|--------|-------|---------|
| number  |       | `rpo`  | TS     | `nua` | TS      |
| string  | `rpo`  |       | `rpo`  | `nua` | TS      |
| BigInt  | TS     | `rpo`  |       | `nua` | TS      |
| any     | `nua`  | `nua`  | `nua`  | `nua` | TS      |
| unknown | TS     | TS     | TS     | TS    | TS      |

Legend:
- `rpo`: The behavior is invalid and should be caught by `restrict-plus-operands`.
- TS: The behavior is invalid and is already caught by Typescript.
- `nua`: The behavior coerces `any` into a stricter type and should thus be caught by `no-unsafe-any`.
- An empty cell means valid behavior.